### ISSUE TITLE
ES6 Map's polyfill isn't installed in phantomjs

### ISSUE THREAD

**Issue by philippjfr** [bokeh/bokeh#8810]
On current master (and 1.1.0rc1) exporting of plot layouts to png using the ``export_png`` function is broken. Here is a reproducible example:

```python
from bokeh.io import export_png

from bokeh.io import output_file, show
from bokeh.layouts import row
from bokeh.plotting import figure

x = list(range(11))
y0 = x
y1 = [10 - i for i in x]
y2 = [abs(i - 5) for i in x]

# create a new plot
s1 = figure(plot_width=250, plot_height=250, title=None)
s1.circle(x, y0, size=10, color="navy", alpha=0.5)

# create another one
s2 = figure(plot_width=250, plot_height=250, title=None)
s2.triangle(x, y1, size=10, color="firebrick", alpha=0.5)

# create and another
s3 = figure(plot_width=250, plot_height=250, title=None)
s3.square(x, y2, size=10, color="olive", alpha=0.5)

export_png(row(s1, s2, s3), filename="plot.png")
```

This errors with the following error:

```
~/miniconda3/envs/pyviz/lib/python3.6/site-packages/PIL/ImageFile.py in _save(im, fp, tile, bufsize)
    493             if o > 0:
    494                 fp.seek(o, 0)
--> 495             e.setimage(im.im, b)
    496             if e.pushes_fd:
    497                 e.setfd(fp)

SystemError: tile cannot extend outside image
```

Presumably because the size of the image is zero, but that's just a guess.

**Comment by bryevdv** [comment]
@philippjfr if you have time to narrow this down some, even just between dev builds if not a full bisection, that would be helpful.

**Comment by mattpap** [comment]
There's also:
```
WARNING:bokeh.io.export:The webdriver raised a TimeoutException while waiting for a 'bokeh:idle' event to signify that the layout has rendered. Something may have gone wrong.
```
the produced image has non-zero size, but is empty, and after cropping its height is reduced to zero. The layout works fine and reports `idle` correctly when used in a standalone HTML file.

**Comment by bryevdv** [comment]
Related question: what is different about this MRE that is not covered by our existing tests (some of which AFAIK generate real PNGs)

**Comment by mattpap** [comment]
Most likely this is a compatibility issue with phantomjs (if true, we will need to IE as well).

**Comment by mattpap** [comment]
Yeah:
```
ReferenceError: Can't find variable: Map
```
That's an easy fix. I will fix #8071 as well, because a proper error should have been reported.
